src/adr/0034-mailbox-api-visibility-and-usage.md

```diff
@@ -0,0 +1,39 @@
+# 34. Mailbox API visibility and usage
+
+Date: 2020-04-27
+
+## Status
+
+Accepted (lazy consensus)
+
+## Context
+
+For some reasons, all mailboxes implementations rely on some abstract classes located in `mailbox-store` module (like 
```

Comment:
mailbox/store defines some common tools to implement mailbox/api. Speaking of abstract classes is maybe a bit too much technical still.